File: objective_c/meow/meow/ViewController.m
//
//  ViewController.m
//  meow
//
//  Created by Gaurav Sara on 10/01/24.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    NSURL *soundURL = [NSURL fileURLWithPath: [[NSBundle mainBundle] pathForResource:@"Cat" ofType:@"wav"]];
    AudioServicesCreateSystemSoundID(( __bridge CFURLRef)soundURL, &soundID);
}


- (IBAction)buttonTapped:(id)sender {
    self.labelView.hidden = NO;
    
    AudioServicesPlaySystemSound(soundID);
    
    [NSTimer scheduledTimerWithTimeInterval:1 target:self selector:@selector(hideLabel) userInfo:nil repeats:NO];
}

- (void)hideLabel {
    self.labelView.hidden = YES;
}
@end

File: objective_c_swift/objective_c_swift/ViewController.swift
//
//  ViewController.swift
//  objective_c_swift
//
//  Created by Gaurav Sara on 10/01/24.
//

import UIKit
import AVFoundation

class ViewController: UIViewController {
    @IBOutlet weak var labelView: UILabel!
    var soundID: SystemSoundID = 0
    
    override func viewDidLoad() {
        super.viewDidLoad()
        if let soundURL = Bundle.main.url(forResource: "Cat", withExtension: "wav") {
            AudioServicesCreateSystemSoundID(soundURL as CFURL, &soundID)
        }
    }

    @IBAction func buttonTapped(_ sender: UIButton) {
        labelView.isHidden = false
        
        AudioServicesPlaySystemSound(soundID)
        
        Timer.scheduledTimer(timeInterval: 1, target: self, selector: #selector(hideLabel), userInfo: nil, repeats: false)
    }
    
    @objc func hideLabel() {
        labelView.isHidden = true
    }
}

